pipeline/trusted-resources/cmd/sign/main.go

```diff
@@ -32,18 +32,22 @@ import (
 	"github.com/sigstore/sigstore/pkg/signature/kms"
 	"github.com/tektoncd/experimental/pipelines/trusted-resources/pkg/trustedtask"
 	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/webhook/resourcesemantics"
 	"sigs.k8s.io/yaml"
 )
 
 var (
-	cosignKey  = flag.String("ck", "", "cosign private key path")
-	kmsKey     = flag.String("kms", "", "kms key path")
-	taskFile   = flag.String("ts", "", "YAML file path for tekton task")
-	targetDir  = flag.String("td", "", "Dir to save the signed files")
-	targetFile = flag.String("tf", "signed.yaml", "Filename of the signed file")
+	cosignKey    = flag.String("ck", "", "cosign private key path")
+	kmsKey       = flag.String("kms", "", "kms key path")
+	resourceFile = flag.String("rf", "", "YAML file path for tekton resources")
+	kind         = flag.String("kd", "Task", "The kind of the signed object")
+	format       = flag.Bool("fmt", true, "Whether we add missing values for the CRD")
```

Comment:
nit: I'd be more descriptive here - e.g. how would a user who's just using the tool without looking at the code would understand what will happen?

- is this just format only changes (i.e. fixing tabs/spaces?)
- what missing values are we adding?

```suggestion
	format       = flag.Bool("default", true, "Whether we add Tekton default values to the CRD before signing")
```

Reply:
+1 for docs

Reply:
Maybe rename this to setdefaults is more clear, format is not so accurate here